frontend/src/pages/DataCatalog/dataset_explorer/DataSourceListing.tsx

```diff
@@ -0,0 +1,173 @@
+import React, { useState } from "react";
+import {
+  DataSourceMetadata,
+  MapOfDatasetMetadata,
+  DatasetMetadata,
+} from "../../../data/utils/DatasetTypes";
+import styles from "./DataSourceListing.module.scss";
+import Button from "@material-ui/core/Button";
+import Card from "@material-ui/core/Card";
+import Link from "@material-ui/core/Link";
+import Typography from "@material-ui/core/Typography";
+import AnimateHeight from "react-animate-height";
+import downloadDataset from "./downloadDataset";
+import DialogTitle from "@material-ui/core/DialogTitle";
+import Dialog from "@material-ui/core/Dialog";
+import List from "@material-ui/core/List";
+import ListItem from "@material-ui/core/ListItem";
+import ListItemIcon from "@material-ui/core/ListItemIcon";
+import CircularProgress from "@material-ui/core/CircularProgress";
+import Alert from "@material-ui/lab/Alert";
+import ListItemText from "@material-ui/core/ListItemText";
+import CheckCircleIcon from "@material-ui/icons/CheckCircle";
+import GetAppIcon from "@material-ui/icons/GetApp";
+import { IconButton } from "@material-ui/core";
+import CloseIcon from "@material-ui/icons/Close";
+
+export type LoadStatus = "loading" | "unloaded" | "error" | "loaded";
+
+function DownloadDatasetListItem(props: {
+  datasetId: string;
+  datasetMetadata: DatasetMetadata;
+}) {
+  const [downloadStatus, setDownloadStatus] = useState<LoadStatus>("unloaded");
+
+  const download = async () => {
+    setDownloadStatus("loading");
+    const state = await downloadDataset(props.datasetId);
+    setDownloadStatus(state ? "loaded" : "error");
+  };
+
+  const getIcon = () => {
+    switch (downloadStatus) {
+      case "unloaded":
+        return <GetAppIcon />;
+      case "loading":
+        return <CircularProgress className={styles.DownloadIcon} />;
+      case "loaded":
+        return <CheckCircleIcon />;
+      default:
+        return "";
+    }
+  };
+
+  if (props.datasetMetadata === undefined) {
+    return <></>;
+  }
```

Comment:
Why allow undefined?

Reply:
In case there is a misconfiguration and there isn't available metadata for a dataset I thought it was better to fail gracefully by just hiding that dataset.

Reply:
Maybe add `getLogger().logError()`? I'm still holding onto hope that one day we'll hook up the logger to real reports that we can monitor on the server

Reply:
good call!